### ISSUE TITLE
Issue after the login oauth2 redirect with PROD profile on angular

### ISSUE THREAD

**Issue by ndywicki** [jhipster/generator-jhipster#23991]
##### **Overview of the issue**

Hello guys,

On the latest beta version v8 of Jhipster, by generating an app with the default parameters and the `OAuth2` and `Angular` options, after to run the app build with `prod` profile (`./mvnw -Pprod`) when the user uses the login, after the IDP redirection (e.g. keycloak) the `Location` parameter of the HTTP response is not correct:

`http://localhost:8080/images/logo-jhipster.png?continue`

instead of: `http://localhost:8080/`

As a result the browser displays an HTTP 403 page:

![jhipster-login-issue](https://github.com/jhipster/generator-jhipster/assets/1721344/a02e606b-f57b-420e-b1d3-36d53abb9eba)

Notes:

* it's ok after manually return on the root url (http://localhost:8080)
* It's ok with dev profile.
* It's ok with react frontend and prod
* It's ok on v7 generator 

##### **Motivation for or Use Case**

Have the OAuth2 OIDC working by default in prod build app.

##### **Reproduce the error**

* Generate JHipster app, choose OAuth2 and Angular options.
* Run Keycloak and PostgreSQL docker containers
* Run the build/app with prod profile with `./mvnw -Pprod`
* Open app and sign-in.

##### **Related issues**

N/A

##### **Suggest a Fix**

N/A

##### **JHipster Version(s)**

`8.0.0-beta.3`

##### **JHipster configuration**

Welcome to JHipster v8.0.0-beta.3

Welcome to the JHipster Information Sub-Generator

```
sample@0.0.1-SNAPSHOT /home/nico/workspaces/jhipster/8.0.0-beta.3/sample
└── generator-jhipster@8.0.0-beta.3
```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "authenticationType": "oauth2",
    "baseName": "sample",
    "buildTool": "maven",
    "cacheProvider": "ehcache",
    "clientFramework": "angular",
    "clientTheme": "none",
    "creationTimestamp": 1698234500263,
    "databaseType": "sql",
    "devDatabaseType": "postgresql",
    "devServerPort": 4200,
    "enableGradleEnterprise": null,
    "enableHibernateCache": true,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "entities": [],
    "gradleEnterpriseHost": null,
    "jhipsterVersion": "8.0.0-beta.3",
    "languages": [
      "en",
      "fr"
    ],
    "messageBroker": false,
    "microfrontend": false,
    "microfrontends": [],
    "nativeLanguage": "en",
    "packageName": "com.mycompany.myapp",
    "prodDatabaseType": "postgresql",
    "reactive": false,
    "searchEngine": false,
    "serverPort": null,
    "serverSideOptions": [],
    "serviceDiscoveryType": false,
    "skipClient": false,
    "skipUserManagement": true,
    "testFrameworks": [],
    "websocket": false,
    "withAdminUi": true
  }
}
</pre>
</details>

##### **Environment and Tools**

openjdk version "17.0.8.1" 2023-08-24
OpenJDK Runtime Environment Temurin-17.0.8.1+1 (build 17.0.8.1+1)
OpenJDK 64-Bit Server VM Temurin-17.0.8.1+1 (build 17.0.8.1+1, mixed mode, sharing)

git version 2.34.1

node: v20.8.0
npm: 10.1.0

Docker version 24.0.5, build ced0996

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

- [x ] Checking this box is mandatory (this is just to show you read everything)

**Comment by mshima** [comment]
I having confirmed the bug but we should switch to use login through ui at cypress.

**Comment by ndywicki** [comment]
I continued to investigate.

I think it's related to the resolution of the relative path of the image by webpack in prod build mode.
In the `loading.css` and the `background-image: url('../images/logo-jhipster.png');`


My hypothesis: since the image is resolved to `http://localhost:8080/image/logo-jhipster.png`, the image is not found, the request go to the backend is as it is the first, it initializes the HTTP session.
Then the request is stored by `HttpSessionRequestCache` (saveRequest).
When signing in Oauth2, after redirection, it is the `saveRequest` which is provided in response to `Location` and the browser continues the redirection to this resource `http://localhost:8080/image/logo-jhipster.png` which does not exist.

The real problem is therefore with webpack build and the assets in prod mode.

As a workaround if I remove the `background-image` from the `loading.css` it's ok.

I keep digging.

**Comment by mraible** [comment]
@ndywicki I've experienced a similar issue with Spring Security SAML and `favicon.ico`. As a workaround, I had to allow access to `favicon.ico` explicitly, and then everything worked as expected. Spring Security ended up fixing it eventually. Here's the issue: https://github.com/spring-projects/spring-security/issues/11657.

**Comment by ndywicki** [comment]
Finally, simply use the asset reference to fix the problem.

@mraible ,Thanks, I see but the security configuration contain already the `content` directory and files extension.
The problem is more linked to the fact that the resource does not exist because the path is not calculated correctly with Angular and the prod mod (see https://github.com/angular/angular-cli/issues/4806)


I created a pull request for a fix : https://github.com/jhipster/generator-jhipster/pull/24047
Testing on Angular, React and Vuejs.

Note: I noticed that the brand logo (jhispster logo by default) does not all have the same rendering/css style.

The cleanest is vueJS:
![jhipster-logo-vuejs](https://github.com/jhipster/generator-jhipster/assets/1721344/855ac556-57ea-4be9-8441-c8308f19f66a)


and the weirdest on ReatJS:
![jhipster-logo-angular](https://github.com/jhipster/generator-jhipster/assets/1721344/3ca3016e-b544-4a1c-ab87-eef328dd0b28)

**Comment by mraible** [comment]
@ndywicki I created apps using the following commands:

```
jhipster --defaults --skip-install --client-framework angular
jhipster --defaults --skip-install --client-framework react
jhipster --defaults --skip-install --client-framework vue
```

Here are the results:

<img width="301" alt="Screenshot 2023-10-30 at 11 44 17 AM" src="https://github.com/jhipster/generator-jhipster/assets/17892/14a1922a-bd81-4287-adba-23554b1c8850">
<img width="335" alt="Screenshot 2023-10-30 at 11 53 44 AM" src="https://github.com/jhipster/generator-jhipster/assets/17892/a587ee58-29ef-4da2-ae9d-700f3ecb0099">
<img width="296" alt="Screenshot 2023-10-30 at 11 56 04 AM" src="https://github.com/jhipster/generator-jhipster/assets/17892/5a1cac93-8c11-4953-b21d-f234f48a5bc2">

I'll attempt to make them more similar.

**Comment by mraible** [comment]
https://github.com/jhipster/generator-jhipster/pull/24050 makes it so the banners are more similar.

**Comment by mraible** [comment]
Closing because it should be fixed in 8.0.0. Please re-open if it doesn't work for you.
